docs/getting_started/go/hello_world_in_go/index.md

```diff
@@ -83,419 +85,778 @@ go get go.temporal.io/sdk
 
-<!--SNIPSTART hello-world-project-template-go-workflow-->
-[workflow.go](https://github.com/temporalio/hello-world-project-template-go/blob/main/workflow.go)
+<!--SNIPSTART go-ipgeo-activity-location-->
+[activities.go](https://github.com/temporalio/temporal-tutorial-ipgeo-go/blob/main/activities.go)
 ```go
-package app
+// GetLocationInfo uses the IP address to fetch location information.
+func (i *IPActivities) GetLocationInfo(ctx context.Context, ip string) (string, error) {
+	url := fmt.Sprintf("http://ip-api.com/json/%s", ip)
+	resp, err := i.HTTPClient.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	var data struct {
+		City       string `json:"city"`
+		RegionName string `json:"regionName"`
+		Country    string `json:"country"`
+	}
+
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%s, %s, %s", data.City, data.RegionName, data.Country), nil
+}
+
+```
+<!--SNIPEND-->
+
+This Activity follows the same pattern as the `getIP` Activity.  This time, the service returns JSON data rather than text, so you have to define a type to unmarshal the data.
+
+While Activities can accept input arguments, it's a best practice to send a single argument rather than multiple arguments. In this case you only have a single String. If you have more than one argument, you should bundle them up in a serializable object. Review the [Activity parameters](https://docs.temporal.io/dev-guide/typescript/foundations/#activity-parameters) section of the Temporal documentation for more details, as there are some limitations you'll want to be aware of in more complex applications.
+
+You've created your two Activities. Now you'll coordinate them using a Temporal Workflow.
+
+## Control application logic with a Workflow
+
+Workflows are where you configure and organize the execution of Activities. You define a Workflow by writing a *Workflow Definition* using one of the Temporal SDKs.
+
+In the Temporal Go SDK, a Workflow Definition is an [exported function](https://go.dev/tour/basics/3) with two additional requirements: it must accept `workflow.Context` as the first input parameter, and it must return `error`. Your Workflow function can optionally return another value, which you'll use to return the result of the Workflow Execution. Review the [Develop Workflows](https://docs.temporal.io/develop/go/core-application#develop-workflows) section of the Temporal documentation for more about Workflows in Go.
+
+Temporal Workflows [must be deterministic](https://docs.temporal.io/workflows#deterministic-constraints) so that Temporal can replay your Workflow in the event of a crash. That's why you call Activities from your Workflow code. Activities don't have the same determinism constraints that Workflows have.
```

Comment:
```suggestion
Temporal Workflows [must be deterministic](https://docs.temporal.io/workflows#deterministic-constraints) so that Temporal can replay your Workflow in the event of a crash. That's why you call Activities from your Workflow code. Unlike Workflows, Activities are not required to be deterministic; they are designed to run steps that can fail.
```